src/yacs-web/src/pages/Main.vue

```diff
@@ -5,110 +5,85 @@
                 <h3>YACS</h3>
                 <hr>
 
-                <div class="course-search">
-                    <b-form-group
-                        label="Search"
-                        label-for="search"
-                    >
-                        <b-form-input id="search" v-model="textSearch" trim placeholder="Intro to College - COLG 1030 - 5/2"></b-form-input>
-                    </b-form-group>
-
-                    <b-form-group
-                        label="Filter Sub-Semester"
-                        for="sub-semester"
-                    >
-                        <b-form-select v-model="selectedSubsemester" :options="subsemesterOptions"></b-form-select>
-                    </b-form-group>
-
-                    <b-form-group
-                        label="Filter Department"
-                        for="department"
-                    >
-                        <b-form-select v-model="selectedDepartment" :options="departmentOptions"></b-form-select>
-                    </b-form-group>
-                </div>
-
-                <hr>
-
-                <b-list-group class="course-list" flush>
-                    <b-list-group-item
-                        v-for="course in filteredCourses"
-                        :key="course.name + course.date_end + course.date_start"
-                        :disabled="course.selected"
-                        :class="{'bg-light': course.selected}"
-                        @click="addCourse(course)"
-                    >
-                        <b>{{ course.name }}</b> ({{ readableDate(course.date_start) }} - {{ readableDate(course.date_end) }}) <br>
-                        {{ course.title }}
-                    </b-list-group-item>
-                </b-list-group>
+                <CourseList
+                    @addCourse="addCourse"
+                />
 
             </b-col>
             <b-col md='8'>
-                <Schedule :courses="selectedCourses" :courseIdentifierFunc="_getCourseIdentifier" @unselectCourse="removeCourse"></Schedule>
+                <b-form-select v-model="selectedScheduleIndex" :options="selectedScheduleOptions"></b-form-select>
+   
+                <Schedule
+                    v-for="(schedule, index) in scheduler.schedules"
```

Comment:
Is it necessary to make separate Schedule components? Makes sense as a mental model, but you can probably get away with one instance and do:

```
                <Schedule
                    v-if="scheduler.schedules[selectedScheduleIndex]"
                    :schedule="scheduler.schedules[selectedScheduleIndex]"
                />
```

Reply:
I have pushed this change to your branch because I could not find a problem with this. Feel free to revert if you know of an issue.

Reply:
Probably insignificant in this use case but my understanding is that `v-if` is conditional rendering while `v-show` is just visibility toggling. So `v-if` will create/destroy the component on every change, but initially will only load one component so "light load, heavy toggle". On the other hand, `v-show` will create and all the components to the DOM on initial load and changes the `display` of components on change so "heavy load, light toggle".
In this case, I'm making the assumption that users are probably going to toggle quite often, so I opt for `v-show`. In my mind, I would probably use `v-if` for say rendering different pages for user and admin. 
The other aspect of it is trying to keep a one-to-one pairing between `ScheduleComponent` and `ScheduleService`. I think this will help with data binding and reactivity but those are my thoughts and I can't come up with a good explanation on why rn lol.
Link to docs: https://vuejs.org/v2/guide/conditional.html#v-if-vs-v-show

Reply:
@123joshuawu Yes, you are correct. I originally threw that on there because the schedule service object was undefined and throwing errors. It would be better to remove that v-if and probably just put a ternary inside the :schedule. Feel free to make that change 🙂

Reply:
Hm interesting did you find out what was causing the service to be undefined?
This is also a sign of my lack of error catching throughout the project atm lol

Reply:
![image](https://user-images.githubusercontent.com/28029704/76689988-3f8ef780-6612-11ea-94ed-91e171ef3cdc.png)

I believe it happens because, in Main.vue, the SubSemesterScheduler is unpopulated with the schedule services by the time `created()` is called on the Schedule component (which depends on the schedule service getter `courseSessionsOnDay`.

While the `created` lifecycle method is synchronous, you're calling `getSubsemester` which is a promise and so that won't halt the Vue lifecycle. Thus, when Schedule.vue's created() is called, its schedule service prop is undefined since it depends on that getSubsemester promise in the parent.

Relevant links: 

- [Async code in created](https://forum.vuejs.org/t/cant-get-created-hook-to-work-when-its-async/28604/4)
- [Parent and child component lifecycle](https://medium.com/@brockreece/vue-parent-and-child-lifecycle-hooks-5d6236bd561f)

Reply:
Interesting... I guess in my head I imagined that the Schedule component would only be created by `v-for` when `SubSemesterScheduler` adds a new `ScheduleService` but I guess I'm missing something

Reply:
My bad, I should have been more specific. I was referring to when `v-for` wasn't on the Schedule component. You are correct when there is a `v-for` attached to it.

As for having multiple schedule components vs one, I didn't think the binding would be that expensive. Feel like the most expensive part is the re-rendering, but the ScheduleEvent is absolutely positioned so it shouldn't be that expensive.

Regardless, it hopefully shouldn't matter much either way. 🤷‍♂

Reply:
Yeah, marking as resolved for now, can open other threads for continuing this line of thought later